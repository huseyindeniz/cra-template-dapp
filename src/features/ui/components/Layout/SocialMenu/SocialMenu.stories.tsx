// ConnectButton.stories.ts|tsx
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { SocialMenu } from './SocialMenu';

export default {
  title: 'ui/Components/Layout/SocialMenu',
  component: SocialMenu,
  args: {
    items: [
      {
        name: 'GitHub',
        link: 'https://github.com/yourusername',
      },
      {
        name: 'Instagram',
        link: 'https://instagram.com/yourusername',
      },
      {
        name: 'Linkedin',
        link: 'https://linkedin.com/yourusername',
      },
      {
        name: 'Twitter',
        link: 'https://twitter.com/yourusername',
      },
      {
        name: 'YouTube',
        link: 'https://youtube.com/yourusername',
      },
      {
        name: 'Discord',
        link: 'https://discord.com/yourusername',
      },
      {
        name: 'Telegram',
        link: 'https://telegram.com/yourusername',
      },
    ],
  },
} as ComponentMeta<typeof SocialMenu>;

export const Default: ComponentStory<typeof SocialMenu> = args => (
  <SocialMenu {...args} />
);
