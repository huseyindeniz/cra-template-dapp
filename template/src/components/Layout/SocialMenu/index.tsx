import React, { ReactNode } from "react";
import { chakra, useColorModeValue, VisuallyHidden } from "@chakra-ui/react";

import { SocialLink, SocialLinkName } from "../../../config/types";

const FaGithub = React.lazy(() =>
  import("@react-icons/all-files/fa/FaGithub").then((module) => ({
    default: module.FaGithub,
  }))
);
const FaTwitter = React.lazy(() =>
  import("@react-icons/all-files/fa/FaTwitter").then((module) => ({
    default: module.FaTwitter,
  }))
);
const FaDiscord = React.lazy(() =>
  import("@react-icons/all-files/fa/FaDiscord").then((module) => ({
    default: module.FaDiscord,
  }))
);
const FaInstagram = React.lazy(() =>
  import("@react-icons/all-files/fa/FaInstagram").then((module) => ({
    default: module.FaInstagram,
  }))
);
const FaLinkedin = React.lazy(() =>
  import("@react-icons/all-files/fa/FaLinkedin").then((module) => ({
    default: module.FaLinkedin,
  }))
);
const FaYoutube = React.lazy(() =>
  import("@react-icons/all-files/fa/FaYoutube").then((module) => ({
    default: module.FaYoutube,
  }))
);
const FaTelegram = React.lazy(() =>
  import("@react-icons/all-files/fa/FaTelegram").then((module) => ({
    default: module.FaTelegram,
  }))
);

const SocialButton = ({
  children,
  label,
  href,
}: {
  children: ReactNode;
  label: string;
  href: string;
}) => {
  return (
    <chakra.button
      bg={useColorModeValue("blackAlpha.100", "whiteAlpha.100")}
      rounded={"full"}
      w={8}
      h={8}
      cursor={"pointer"}
      as={"a"}
      href={href}
      target={"_blank"}
      display={"inline-flex"}
      alignItems={"center"}
      justifyContent={"center"}
      transition={"background 0.3s ease"}
      _hover={{
        bg: useColorModeValue("blackAlpha.200", "whiteAlpha.200"),
      }}
    >
      <VisuallyHidden>{label}</VisuallyHidden>
      {children}
    </chakra.button>
  );
};

const Icon = ({ name }: { name: SocialLinkName }) => {
  switch (name) {
    case "GitHub":
      return <FaGithub />;
    case "Instagram":
      return <FaInstagram />;
    case "Linkedin":
      return <FaLinkedin />;
    case "Twitter":
      return <FaTwitter />;
    case "YouTube":
      return <FaYoutube />;
    case "Discord":
      return <FaDiscord />;
    case "Telegram":
      return <FaTelegram />;
  }
};

export interface SocialMenuProps {
  items: SocialLink[];
}

export const SocialMenu: React.FC<SocialMenuProps> = React.memo(({ items }) => {
  return (
    <>
      {items &&
        items.map((i) => (
          <SocialButton label={i.name} href={i.link} key={i.name}>
            <Icon name={i.name} />
          </SocialButton>
        ))}
    </>
  );
});
