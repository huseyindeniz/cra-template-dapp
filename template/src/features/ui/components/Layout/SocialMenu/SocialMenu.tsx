import {
  chakra,
  Stack,
  useColorModeValue,
  VisuallyHidden,
} from '@chakra-ui/react';
import React, { ReactNode } from 'react';

const FaGithub = React.lazy(() =>
  import('@react-icons/all-files/fa/FaGithub').then(module => ({
    default: module.FaGithub,
  }))
);
const FaLinkedin = React.lazy(() =>
  import('@react-icons/all-files/fa/FaLinkedin').then(module => ({
    default: module.FaLinkedin,
  }))
);

const SocialButton = ({
  children,
  label,
  href,
}: {
  children: ReactNode;
  label: string;
  href: string;
}) => {
  return (
    <chakra.button
      bg={useColorModeValue('blackAlpha.100', 'whiteAlpha.100')}
      rounded="full"
      role="button"
      w={8}
      h={8}
      cursor="pointer"
      as="a"
      href={href}
      target="_blank"
      display="inline-flex"
      alignItems="center"
      justifyContent="center"
      transition="background 0.3s ease"
      _hover={{
        bg: useColorModeValue('blackAlpha.200', 'whiteAlpha.200'),
      }}
    >
      <VisuallyHidden>{label}</VisuallyHidden>
      {children}
    </chakra.button>
  );
};

export const SocialMenu: React.FC = React.memo(() => {
  return (
    <Stack direction="row" spacing={6}>
      <SocialButton
        label="GitHub"
        href="https://github.com/huseyindeniz"
        key="GitHub"
      >
        <React.Suspense fallback="">
          <FaGithub />
        </React.Suspense>
      </SocialButton>
      <SocialButton
        label="Linkedin"
        href="https://linkedin.com/in/huseyindenizkivrak/en-us"
        key="Linkedin"
      >
        <React.Suspense fallback="">
          <FaLinkedin />
        </React.Suspense>
      </SocialButton>
    </Stack>
  );
});
