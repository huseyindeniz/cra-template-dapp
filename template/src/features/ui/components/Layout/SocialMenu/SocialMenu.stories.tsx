// ConnectButton.stories.ts|tsx
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { SocialMenu } from './SocialMenu';

export default {
  title: 'ui/Components/Layout/SocialMenu',
  component: SocialMenu,
} as ComponentMeta<typeof SocialMenu>;

const Template: ComponentStory<typeof SocialMenu> = args => (
  <SocialMenu {...args} />
);

export const Default = Template.bind({});
