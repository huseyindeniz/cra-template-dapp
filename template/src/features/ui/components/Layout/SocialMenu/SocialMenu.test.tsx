import { composeStories } from '@storybook/testing-react';
import { render, act } from '@testing-library/react';
import React from 'react';

import '@testing-library/jest-dom/extend-expect';

import * as stories from './SocialMenu.stories';

describe('Feature: UI', () => {
  describe('Component: Layout/SocialMenu', () => {
    const { Default } = composeStories(stories);
    describe('Scenario: Default', () => {
      it('renders a GitHub and LinkedIn button', async () => {
        // Arrange + Act
        const { asFragment, getByRole } = await act(async () =>
          render(<Default {...Default.args} />)
        );
        // Assert
        expect(getByRole('button', { name: 'GitHub' })).toBeInTheDocument();
        expect(getByRole('button', { name: 'Linkedin' })).toBeInTheDocument();
        expect(asFragment).toMatchSnapshot();
      });
    });
  });
});
