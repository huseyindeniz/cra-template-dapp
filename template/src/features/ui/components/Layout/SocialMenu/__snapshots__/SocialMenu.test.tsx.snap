// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: UI Component: Layout/SocialMenu Scenario: Default renders a GitHub and LinkedIn button 1`] = `[Function]`;
